[pt-br] Add docs/tasks/access-application-cluster/access-cluster-services.md (#40546)
kubernetes/website: content/pt-br/docs/tasks/access-application-cluster/access-cluster-services.md

---
title: Acessando serviços em execução em clusters
content_type: task
weight: 140
---

<!-- overview -->
Esta página mostra como se conectar aos serviços em execução no cluster Kubernetes.

> {{% heading "prerequisites" %}}

Reviewer: ```suggestion
## {{% heading "prerequisites" %}}
```

Missing the title markdown here.

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- steps -->

## Acessando serviços em execução no cluster

> No Kubernetes, [nós](/pt-br/docs/concepts/architecture/nodes/), [pods](/docs/concepts/workloads/pods/) e [serviços](/docs/concepts/services-networking/service/) têm todos seus próprios IPs. Em muitos casos, os IPs dos nós, dos pods e alguns dos IPs de serviço em um cluster não serão

Reviewer: ```suggestion
No Kubernetes, [nós](/pt-br/docs/concepts/architecture/nodes/), [Pods](/docs/concepts/workloads/pods/) e [serviços](/docs/concepts/services-networking/service/) têm todos seus próprios IPs. Em muitos casos, os IPs dos nós, dos Pods e alguns dos IPs de serviço em um cluster não serão
```
Minor, but we should always render Pod with a capital P.

> No Kubernetes, [nós](/pt-br/docs/concepts/architecture/nodes/), [Pods](/docs/concepts/workloads/pods/) e [serviços](/docs/concepts/services-networking/service/) têm todos seus próprios IPs. Em muitos casos, os IPs dos nós, dos Pods e alguns dos IPs de serviço em um cluster não serão

Reviewer 2: Sounds better to me "No Kubernetes, todos nós, Pods e serviços têm seus próprios IPs." instead of "No Kubernetes, nós, Pods e serviços todos têm seus próprios IPs.". Think about it.

> roteáveis, portanto, não estarão acessíveis a partir de uma máquina fora do cluster, como seu computador de mesa.

Reviewer: "Computador de mesa" sounds a little weird to me. Perhaps one of "computador pessoal", "estação de trabalho", or even just "computador" could fit better here?

### Maneiras de se conectar

> Você tem várias opções para se conectar a nós, pods e serviços de fora do cluster:

Reviewer 2: "Pods" instead of "pods"

  - Acesse serviços através de IPs públicos.
    - Use um serviço com tipo `NodePort` ou `LoadBalancer` para tornar o serviço acessível fora do cluster. Consulte a documentação de [serviços](/docs/concepts/services-networking/service/) e
[kubectl expose](/docs/reference/generated/kubectl/kubectl-commands/#expose).
    - Dependendo do ambiente do cluster, isso pode expor o serviço apenas para a rede corporativa, ou pode expô-lo para a Internet. Pense se o serviço que está sendo exposto é seguro. Ele faz sua própria autenticação?

>     - Coloque Pods atrás de serviços. Para acessar um Pod específico de um conjunto de réplicas, como para depurar, coloque um rótulo exclusivo no Pod e crie um novo serviço que selecione este rótulo.

Reviewer 2: "coloque uma label exclusiva no Pod e ... selecione esta label." instead of "coloque um rótulo exclusivo no Pod e ... selecione este rótulo.". In this context, the "label" is a key-value pair.

>     - Na maioria dos casos, não deve ser necessário para o desenvolvedor de aplicativos acessar diretamente nós através de seus IPs de nó.

Reviewer: ```suggestion
    - Na maioria dos casos, não deve ser necessário para o desenvolvedor de aplicativos acessar diretamente nós através de seus endereços IP.
```

I suspect `nodeIPs` here referred to a field. I don't think it's necessary to reiterate that it's a node IP.

  - Acesse serviços, nós ou Pods usando o Verbo Proxy.
    - Faz autenticação e autorização do servidor de API antes de acessar o serviço remoto. Use isto se os serviços não forem seguros o suficiente para expor à Internet, ou para obter acesso a portas no IP do nó, ou para depuração.
    - Proxies podem causar problemas para algumas aplicações web.
    - Só funciona para HTTP/HTTPS.
    - Descrito [aqui](#manually-constructing-apiserver-proxy-urls).
  - Acesse a partir de um nó ou Pod no cluster.
    - Execute um Pod e, em seguida, conecte-se a um shell nele usando [kubectl exec](/docs/reference/generated/kubectl/kubectl-commands/#exec). Conecte-se a outros nós, Pods e serviços a partir desse shell.

>     - Alguns clusters podem permitir que você faça ssh para um nó no cluster. De lá, você pode conseguir acessar os serviços do cluster. Este é um método não padrão e funcionará em alguns clusters, mas não em outros. Navegadores e outras ferramentas podem ou não estar instalados. O DNS do cluster pode não funcionar.

Reviewer 2: Sounds better to me "Este é um método que não é padrão" instead of "Este é um método não padrão".

>     - Alguns clusters podem permitir que você faça ssh para um nó no cluster. De lá, você pode ser capaz de acessar os serviços do cluster. Este é um método não padrão e funcionará em alguns clusters, mas não em outros. Navegadores e outras ferramentas podem ou não estar instalados. O DNS do cluster pode não funcionar.

Reviewer: ```suggestion
    - Alguns clusters podem permitir que você faça ssh para um nó no cluster. De lá, você pode conseguir acessar os serviços do cluster. Este é um método não padrão e funcionará em alguns clusters, mas não em outros. Navegadores e outras ferramentas podem ou não estar instalados. O DNS do cluster pode não funcionar.
```

I think "conseguir" sounds a bit better than "ser capaz" in this context.

### Descobrindo serviços integrados

Normalmente, existem vários serviços que são iniciados em um cluster pelo kube-system. Obtenha uma lista desses serviços com o comando `kubectl cluster-info`:

```shell
kubectl cluster-info
```

A saída é semelhante a esta:

```
Kubernetes master is running at https://192.0.2.1
elasticsearch-logging is running at https://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy
kibana-logging is running at https://192.0.2.1/api/v1/namespaces/kube-system/services/kibana-logging/proxy
kube-dns is running at https://192.0.2.1/api/v1/namespaces/kube-system/services/kube-dns/proxy
grafana is running at https://192.0.2.1/api/v1/namespaces/kube-system/services/monitoring-grafana/proxy
heapster is running at https://192.0.2.1/api/v1/namespaces/kube-system/services/monitoring-heapster/proxy
```

> Isso mostra a URL proxy-verbo para acessar cada serviço. Por exemplo, este cluster tem o registro em nível de cluster habilitado (usando o Elasticsearch), que pode ser acessado em `https://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/` se as credenciais adequadas forem passadas ou através de um proxy kubectl, por exemplo: `http://localhost:8080/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/`.

Reviewer: ```suggestion
Isso mostra a URL referente ao verbo proxy para acessar cada serviço. Por exemplo, este cluster tem _logs_ a nível de cluster habilitados (usando o Elasticsearch), que pode ser acessado em `https://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/` se as credenciais adequadas forem passadas ou através do comando kubectl proxy, como por exemplo: `http://localhost:8080/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/`.
```

If I understood correctly, the intention here was to tell the user they can proxy the Kubernetes API using `kubectl proxy`, then access the service using the given URL.

{{< note >}}

> Consulte [Acessando clusters usando a API Kubernetes](/docs/tasks/administer-cluster/access-cluster-api/#accessing-the-cluster-api) para obter informações sobre como passar credenciais ou usar o proxy kubectl.

Reviewer: ```suggestion
Consulte [Acessando clusters usando a API do Kubernetes](/docs/tasks/administer-cluster/access-cluster-api/#accessing-the-cluster-api) para obter informações sobre como passar credenciais ou usar o comando kubectl proxy.
```

Same consideration here as the previous paragraph.

{{< /note >}}

> ### Construindo manualmente URLs de proxy do apiserver

Reviewer: ```suggestion
### Construindo manualmente URLs de proxy do servidor da API {##manually-constructing-apiserver-proxy-urls}
```

Suggestion: change "apiserver" for "servidor da API". We also need an anchor here so that the link on line 35 works.

> Como mencionado acima, você usa o comando `kubectl cluster-info` para recuperar a URL do proxy do serviço. Para criar URLs de proxy que incluem pontos de extremidade, sufixos e parâmetros de serviço, você adiciona à URL do proxy do serviço:
> "`http://`*`endereço_do_mestre_do_kubernetes`*`/api/v1/namespaces/`*`nome_do_namespace`*`/services/`[https:]nome_do_serviço[:nome_da_porta]`*`/proxy`"

Reviewer: ```suggestion
Como mencionado acima, você usa o comando `kubectl cluster-info` para recuperar a URL do proxy do serviço. Para criar URLs de proxy que incluem _endpoints_, sufixos e parâmetros de serviço, você adiciona à URL do proxy do serviço:
`http://`*`endereço_do_mestre_do_kubernetes`*`/api/v1/namespaces/`*`nome_do_namespace`*`/services/`*`[https:]nome_do_serviço[:nome_da_porta]`*`/proxy`
```

I'm unfamiliar with a good translation of endpoint to Portuguese. I lean towards keeping it as _endpoint_, since it's well-known terminology for APIs.

There were also no enclosing quotes in the original text, and there's a missing few format signs that's currently breaking rendering.

Se você não especificou um nome para a porta, não é necessário especificar *nome_da_porta* na URL. Você também pode usar o número da porta no lugar do *nome_da_porta* para portas nomeadas e não nomeadas.

> Por padrão, o servidor API usa proxy para o seu serviço usando HTTP. Para usar HTTPS, adicione o prefixo `https:` ao nome do serviço:
> `http://<endereço_do_mestre_do_kubernetes>/api/v1/namespaces/<nome_do_namespace>/services/<nome_do_serviço>/proxy`
> 
> Os formatos suportados para o segmento `<nome_do_serviço>` da URL são:
> 
> * `<nome_do_serviço>` - faz proxy para a porta padrão ou não nomeada usando http
> * `<nome_do_serviço>:<nome_da_porta>` - faz proxy para a porta nomeada ou número da porta especificado usando http
> * `https:<nome_do_serviço>:` - faz proxy para a porta padrão ou não nomeada usando https (observe o dois-pontos no final)
> * `https:<nome_do_serviço>:<nome_da_porta>` - faz proxy para a porta nomeada ou número da porta especificado usando https

Reviewer: ```suggestion
Por padrão, o servidor da API usa um proxy para o seu serviço através de HTTP. Para usar HTTPS, adicione o prefixo `https:` ao nome do serviço:
`http://<endereço_do_mestre_do_kubernetes>/api/v1/namespaces/<nome_do_namespace>/services/<nome_do_serviço>/proxy`

Os formatos suportados para o segmento `<nome_do_serviço>` da URL são:

* `<nome_do_serviço>` - usa um proxy para a porta padrão ou não nomeada usando http
* `<nome_do_serviço>:<nome_da_porta>` - usa um proxy para a porta nomeada ou número da porta especificado usando http
* `https:<nome_do_serviço>:` - usa um proxy para a porta padrão ou não nomeada usando https (observe o dois-pontos no final)
* `https:<nome_do_serviço>:<nome_da_porta>` - usa um proxy para a porta nomeada ou número da porta especificado usando https
```

Some minor suggestions here.

##### Exemplos

> * Para acessar o ponto de extremidade de serviço Elasticsearch `_search?q=user:kimchy`, você usaria:

Reviewer: ```suggestion
* Para acessar o _endpoint_ de serviço Elasticsearch `_search?q=user:kimchy`, você usaria:
```

  ```
  http://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/_search?q=user:kimchy
  ```
  

> * Para acessar as informações de saúde do cluster Elasticsearch `_cluster/health?pretty=true`, você usaria:

Reviewer: ```suggestion
* Para acessar as informações de integridade do cluster Elasticsearch `_cluster/health?pretty=true`, você usaria:
```

"Saúde" seems to refer to a human, but this is a machine. So perhaps integridade is a better textual fit.

  ```
  https://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/_cluster/health?pretty=true
  ```

>   As informações de saúde são semelhantes a estas:

Reviewer: ```suggestion
  As informações de integridade são semelhantes a estas:
```

  
  ```json
    {
      "cluster_name" : "kubernetes_logging",
      "status" : "yellow",
      "timed_out" : false,
      "number_of_nodes" : 1,
      "number_of_data_nodes" : 1,
      "active_primary_shards" : 5,
      "active_shards" : 5,
      "relocating_shards" : 0,
      "initializing_shards" : 0,
      "unassigned_shards" : 5
    }
    ```
    

> * Para acessar as informações de saúde do serviço Elasticsearch `_cluster/health?pretty=true`, você usaria:

Reviewer: ```suggestion
* Para acessar as informações de integridade do serviço Elasticsearch `_cluster/health?pretty=true`, você usaria:
```

  ```
  https://192.0.2.1/api/v1/namespaces/kube-system/services/https:elasticsearch-logging:/proxy/_cluster/health?pretty=true
  ```

> #### Usando navegadores da web para acessar serviços em execução no cluster
> 
> Você pode ser capaz de colocar um URL de proxy de apiserver na barra de endereço de um navegador. No entanto:
> 
>   - Os navegadores da web geralmente não podem passar tokens, portanto, você pode precisar usar autenticação básica (senha). Apiserver pode ser configurado para aceitar autenticação básica, mas o seu cluster pode não estar configurado para aceitar autenticação básica.
>   - Alguns aplicativos da web podem não funcionar, principalmente aqueles com javascript do lado do cliente que constroem URLs de uma maneira que não está ciente do prefixo do caminho do proxy.

Reviewer: ```suggestion
#### Usando navegadores da web para acessar serviços em execução no cluster

Você pode conseguir de colocar um URL de proxy do servidor da API na barra de endereço de um navegador. No entanto:

  - Os navegadores da web geralmente não podem passar tokens, portanto, você pode precisar usar autenticação básica (senha). O servidor da API pode ser configurado para aceitar autenticação básica, mas o seu cluster pode não estar configurado para aceitar autenticação básica.
  - Alguns aplicativos da web podem não funcionar, principalmente aqueles com javascript do lado do cliente que constroem URLs com um mecanismo que não está ciente do prefixo do caminho do proxy.
```

Some more textual suggestions.

>   - Alguns aplicativos da web podem não funcionar, principalmente aqueles com javascript do lado do cliente que constroem URLs com um mecanismo que não está ciente do prefixo do caminho do proxy.

Reviewer 2: You used "aplicações" before (line 33) and I liked it. 
"Algumas aplicações da web" instead of "Alguns aplicativos da web"

---
title: Acessando serviços em execução em clusters
content_type: task
weight: 140
---

<!-- overview -->

> Esta página mostra como se conectar a serviços em execução no cluster Kubernetes.

Reviewer 2: Sounds better to me "se conectar aos serviços em execução" instead of "se conectar a serviços em execução". Think about it.

> {{% heading "pré-requisitos" %}}

Reviewer 2: "prerequisites" instead of "pré-requisitos"

{{< include "task-tutorial-prereqs.md" >}} {{< version-check >}}

<!-- steps -->

## Acessando serviços em execução no cluster

> No Kubernetes, [nós](/docs/concepts/architecture/nodes/), [pods](/docs/concepts/workloads/pods/) e [serviços](/docs/concepts/services-networking/service/) têm todos seus próprios IPs. Em muitos casos, os IPs dos nós, dos pods e alguns dos IPs de serviço em um cluster não serão

Reviewer 2: Add "/pt-br/" at nodes link

roteáveis, portanto, não estarão acessíveis a partir de uma máquina fora do cluster, como seu computador de mesa.

> ## Maneiras de se conectar

Reviewer 2: The original file is "###"

Você tem várias opções para se conectar a nós, pods e serviços de fora do cluster:

  - Acesse serviços através de IPs públicos.

>     - Use um serviço com tipo `NodePort` ou `LoadBalancer` para tornar o serviço acessível fora do cluster. Consulte a documentação de [serviços]((/docs/concepts/services-networking/service/)) e

Reviewer 2: The link is broken. Remove the double "("

[kubectl expose](/docs/reference/generated/kubectl/kubectl-commands/#expose).
    - Dependendo do ambiente do cluster, isso pode expor o serviço apenas para a rede corporativa, ou pode expô-lo para a Internet. Pense se o serviço que está sendo exposto é seguro. Ele faz sua própria autenticação?

>     - Coloque pods atrás de serviços. Para acessar um pod específico de um conjunto de réplicas, como para depurar, coloque um rótulo exclusivo no pod e crie um novo serviço que selecione este rótulo.

Reviewer 2: "Pod" instead of "pod". 

We standardized use of Pod with a capital letter. Check below

    - Na maioria dos casos, não deve ser necessário para o desenvolvedor de aplicativos acessar diretamente nós através de seus IPs de nó.
  - Acesse serviços, nós ou pods usando o Verbo Proxy.
    - Faz autenticação e autorização do servidor de API antes de acessar o serviço remoto. Use isto se os serviços não forem seguros o suficiente para expor à Internet, ou para obter acesso a portas no IP do nó, ou para depuração.
    - Proxies podem causar problemas para algumas aplicações web.
    - Só funciona para HTTP/HTTPS.
    - Descrito [aqui](#manually-constructing-apiserver-proxy-urls).
  - Acesse a partir de um nó ou pod no cluster.
    - Execute um pod e, em seguida, conecte-se a um shell nele usando [kubectl exec](/docs/reference/generated/kubectl/kubectl-commands/#exec). Conecte-se a outros nós, pods e serviços a partir desse shell.
    - Alguns clusters podem permitir que você faça ssh para um nó no cluster. De lá, você pode ser capaz de acessar os serviços do cluster. Este é um método não padrão e funcionará em alguns clusters, mas
não em outros. Navegadores e outras ferramentas podem ou não estar instalados. O DNS do cluster pode não funcionar.

> ## Descobrindo serviços integrados

Reviewer 2: The original file is "###"

Normalmente, existem vários serviços que são iniciados em um cluster pelo kube-system. Obtenha uma lista desses serviços com o comando `kubectl cluster-info`:

```shell
kubectl cluster-info
```

A saída é semelhante a esta:

```
O mestre do Kubernetes está em execução em https://192.0.2.1
elasticsearch-logging está em execução em https://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy
kibana-logging está em execução em https://192.0.2.1/api/v1/namespaces/kube-system/services/kibana-logging/proxy
kube-dns está em execução em https://192.0.2.1/api/v1/namespaces/kube-system/services/kube-dns/proxy
grafana está em execução em https://192.0.2.1/api/v1/namespaces/kube-system/services/monitoring-grafana/proxy

> heapster está em execução em https://192.0.2.1/api/v1/namespaces/kube-system/services/monitoring-heapster/proxy

Reviewer 2: I liked it "Mestre do Kubernetes", but command output it is better keep the original output.

```

Isso mostra a URL proxy-verbo para acessar cada serviço. Por exemplo, este cluster tem o registro em nível de cluster habilitado (usando o Elasticsearch), que pode ser acessado em `https://192.0.2.1/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/` se as credenciais adequadas forem passadas ou através de um proxy kubectl, por exemplo: `http://localhost:8080/api/v1/namespaces/kube-system/services/elasticsearch-logging/proxy/`.

{{< note >}}
Consulte [Acessando clusters usando a API Kubernetes](/docs/tasks/administer-cluster/access-cluster-api/#accessing-the-cluster-api) para obter informações sobre como passar credenciais ou usar o proxy kubectl.
{{< /note >}}

> ## Construindo manualmente URLs de proxy do apiserver

Reviewer 2: The original file is "###"

Como mencionado acima, você usa o comando `kubectl cluster-info` para recuperar a URL do proxy do serviço. Para criar URLs de proxy que incluem pontos de extremidade, sufixos e parâmetros de serviço, você adiciona à URL do proxy do serviço:

> `http://`*`endereço_do_mestre_do_kubernetes`*`/api/v1/namespaces/`*`nome_do_namespace`*`/services/`[https:]nome_do_serviço[:nome_da_porta]`*`/proxy`

Reviewer 2: Minor: add "*`" before "[https:]service_name[:port_name]`*`/proxy`"
